### ISSUE TITLE
one more bug in population across several connections

### ISSUE THREAD

**Issue by TrejGun** [Automattic/mongoose#4312]
very similar to #4243

case 1
```js
var mongoose = require("mongoose");

var db1 = mongoose.createConnection();
db1.open("mongodb://localhost/issue-XXXX-1");
var db2 = mongoose.createConnection();
db2.open("mongodb://localhost/issue-XXXX-2");

var Z = db2.model("Z", new mongoose.Schema({
	name: String
}));
var Y = db1.model("Y", new mongoose.Schema({
	name: String,
	z: {
		type: mongoose.Schema.Types.ObjectId,
		// ref: "Z" // <- uncomment me
	}
}));
var X = db1.model("X", new mongoose.Schema({
	name: String,
	y: {
		type: mongoose.Schema.Types.ObjectId,
		ref: "Y"
	}
}));


var z = new Z({
	name: "Z"
});
z.save(function (err, z) {
	var y = new Y({
		name: "Y",
		z: z
	});
	y.save(function (err, y) {
		var x = new X({
			name: "X",
			y: y
		});
		x.save(function (err, x) {

			X.findById(x).populate({
				path: "y",
				model: db1.model("Y"),
				populate: [{
					path: "z",
					model: db2.model("Z")
				}]
			}).exec(function (err, xx) {
				console.log("xx", xx)
			});


		});
	});
});
```

case 2
```js
var mongoose = require("mongoose");

var db1 = mongoose.createConnection();
db1.open("mongodb://localhost/issue-XXXX-1");
var db2 = mongoose.createConnection();
db2.open("mongodb://localhost/issue-XXXX-2");

var Z = db1.model("Z", new mongoose.Schema({
	name: String
}));
var Y = db2.model("Y", new mongoose.Schema({
	name: String,
	z: {
		type: mongoose.Schema.Types.ObjectId,
		ref: "Z"
	}
}));
var X = db1.model("X", new mongoose.Schema({
	name: String,
	y: {
		type: mongoose.Schema.Types.ObjectId,
		// ref: "Y" // <- uncomment me
	}
}));


var z = new Z({
	name: "Z"
});
z.save(function (err, z) {
	var y = new Y({
		name: "Y",
		z: z
	});
	y.save(function (err, y) {
		var x = new X({
			name: "X",
			y: y
		});
		x.save(function (err, x) {

			X.findById(x).populate({
				path: "y",
				model: db2.model("Y"),
				populate: [{
					path: "z",
					model: db1.model("Z")
				}]
			}).exec(function (err, xx) {
				console.log("xx", xx)
			});


		});
	});
});
```

in both cases error is
```
MissingSchemaError: Schema hasn't been registered for model "M".
```
this was working before

**Comment by TrejGun** [comment]
@vkarpov15 
looks like this was fixed in 76ea42508b62000c7a8e92b4b021de926c8f2218
Thank you so much for you work!

**Comment by vkarpov15** [comment]
Thank goodness. Looks like that `subPopulate()` function was causing a lot of issues
